src/s1reader/s1_annotation.py

```diff
@@ -7,20 +7,75 @@
 
 import datetime
 import os
-import lxml.etree as ET
 import zipfile
 
+from types import SimpleNamespace
+
+import lxml.etree as ET
 import numpy as np
 
-from scipy.interpolate import InterpolatedUnivariateSpline, interp1d
+from isce3.core import speed_of_light
 from packaging import version
+from scipy.interpolate import InterpolatedUnivariateSpline, interp1d
 
-from isce3.core import speed_of_light
 
 # Minimum IPF version from which the S1 product's Noise Annotation
 # Data Set (NADS) includes azimuth noise vector annotation
 min_ipf_version_az_noise_vector = version.parse('2.90')
 
+# Minimum IPF version from which the RFI information gets available
+RFI_INFO_AVAILABLE_FROM = version.Version('3.40')
+
+# Dictionary of the fields in RFI information, and their data type castor
+dict_datatype_rfi={
+    "swath": str,
+    "azimuthTime": lambda T: datetime.datetime.strptime(T, '%Y-%m-%dT%H:%M:%S.%f'),
+    "inBandOutBandPowerRatio": float,
+    "percentageAffectedLines": float,
+    "avgPercentageAffectedSamples": float,
+    "maxPercentageAffectedSamples": float,
+    "numSubBlocks": int,
+    "subBlockSize": int,
+    "maxPercentageAffectedBW": float,
+    "percentageBlocksPersistentRfi": float,
```

Comment:
```suggestion
    "percentageBlocksPersistentRFI": float,
```
For consistency with Gustavo's comment.

Reply:
The keys in `dict_datatype_rfi` have to be in exact match with the field names in RFI annotation .xml files because the following function `element_to_dict()` make use of the dictionary to parse the XML.